File: docs/courses/data-structure/samples/queue/queueArray.py
class queueArray:

    def __init__(self,size):
        self.queue = list(0 for i in range(size)) ## 큐 사이즈 지정
        self.front = -1
        self.rear = -1
        self.size = size

    def printQueue(self):

        for i in range(0,self.size):
            if (i<=self.front) or (i > self.rear):
                print(" ")
            else:
                print(self.queue[i], end=' ')

        print("\n")


    def isEmpty(self):

        if self.front == self.rear:
            return True
        else:
            return False

    def isFull(self):

        if self.rear == (self.size - 1):
            return True
        else :
             return False

    def enqueue(self,value):
        if self.isFull():
            print("큐가 가득찼습니다.")
            return
        else:
            self.rear = self.rear + 1
            self.queue[self.rear] = value

    def dequeue(self):
        if self.isEmpty():
            print("큐가 비었습니다.")
            return

        else:
            self.front = self.front + 1
            result = self.queue[self.front]
            return result


if __name__ == '__main__':

    queue = queueArray(5)

    queue.enqueue(1)
    queue.printQueue()  ## 1

    queue.enqueue(2)
    queue.printQueue()  ## 1 2

    queue.enqueue(3)
    queue.printQueue()  ## 1 2 3

    queue.enqueue(4)
    queue.printQueue()  ## 1 2 3 4

    queue.enqueue(5)
    queue.printQueue()  ## 1 2 3 4 5

    queue.enqueue(6)
    queue.printQueue()  ## full

    queue.dequeue()
    queue.printQueue()  ## 2 3 4 5

File: docs/courses/data-structure/samples/queue/queueArray.java
package DataStructure;

public class queueArray {

    private int[] queue;

    private int front;
    private int rear;

    public queueArray(int size){ // 큐 생성자

        this.queue = new int[size];
        this.front = -1;
        this.rear = -1;

    }

    void printQueue(){

        for(int i=0; i<queue.length; i++){
            if( i<=front || i>rear)
                System.out.print(" ");
            else
                System.out.print(queue[i] + " ");

        }
        System.out.println();
    }

    boolean isEmpty(){
        if(front == rear){
            return true;
        }
        else{
            return false;
        }
    }

    boolean isFull(){
        if(rear == queue.length-1){
            return true;
        }
        else{
            return false;
        }
    }

    void enqueue(int value){
        if(isFull()){
            System.out.println("큐가 가득찼습니다.");
        }
        else{
            queue[++rear] = value;
        }
    }

    int dequeue(){
        if(isEmpty()){
            System.out.println("큐가 비었습니다.");
            return -1;
        }
        else{
            int result = queue[++front];
            return result;
        }
    }


    public static void main(String[] args) {

        queueArray queue = new queueArray(5);

        queue.enqueue(1);
        queue.printQueue();

        queue.enqueue(2);
        queue.printQueue();

        queue.enqueue(3);
        queue.printQueue();

        queue.enqueue(4);
        queue.printQueue();

        queue.enqueue(5);
        queue.printQueue();

        queue.enqueue(6);

        queue.dequeue();
        queue.printQueue();

    }
}



